### ISSUE TITLE
Please add a WATCHDOG 

### ISSUE THREAD

**Issue by itaosson** [ethereum-mining/ethminer#327]
Please increase the watchdog,when  cuda crash and something errors. we need
Automatic restart the miner.

**Comment by mikebzh44** [comment]
+1

I'm using Claymore but Ethminer give me more hashrate. But I can't use it, it's not stable (lost GPU, ...).

Please add watchdog like in Claymore to specifie  a minimum hashrate and to reboot miner and / or computer.

Thanks.

**Comment by piotr-dobrogost** [comment]
Duplicate of issue #274.

**Comment by ykhuat** [comment]
Maybe you can have a look here.
Auto restart ethminer if no job for 5 min time.
Auto restart system if "CUDA ERROR" detected.
https://bitcointalk.org/index.php?topic=2195527.0

**Comment by marcofranssen** [comment]
+1 

Also see #340.

**Comment by ThetaProtocol** [comment]
https://ethmonitoring.com/
Been using there EthControl program the past two weeks acts as watchdog, allows for RC from anywhere including app. Has support for ethminer,eminer, claymore and more for other crypto. bonus uptime, pool stats+ profit charts. Downside is for more than one rig requires premuim, which maxes out at 99 rigs, which new accounts get 5 days free  which I find it worth it due to ease of use + etra features and you do get notifications on rigs you can customize

**Comment by bmatthewshea** [comment]
If you run Linux you can do something like this in the meantime -
https://github.com/ethereum-mining/ethminer/issues/310#issuecomment-336573725
And use monit or similar to monitor it/restart service/etc by simply monitoring the output (or lack of).

**Comment by jaddison** [comment]
Not sure if this works on Windows (I've only tested on Ubuntu 16.04), but see my gist here https://gist.github.com/jaddison/d4966f9e789d1d6e987fda49638a284b

It's an `ethminer` process wrapper to monitor output and restart as necessary. Lets you increase overclock settings a bit more as it will restart on `CUDA error` output.

**Comment by one-quaker** [comment]
Some watchdogs you can find here (ethminer, ewbf, gpu load) [https://github.com/one-quaker/linux-rig-tools](url)
development in progress

**Comment by DeadManWalkingTO** [comment]
After #757 (added --exit parameter to exit whenever an error occurred) you can use a watchdog.

Here is my [ETHminerWatchDogDmW](https://github.com/DeadManWalkingTO/ETHminerWatchDogDmW) Windows7/8/10 [32/64] & Linux (Any Dist/Any Ver/Any Arch) (#735).

Check and feedback please.
Thank you!

**Comment by Undertrey** [comment]
Also +1 open source BATCH watchdog script for Windows users:
https://github.com/Undertrey/FarmWatchBot/releases

**Comment by XhmikosR** [comment]
Duplicate of #274
